database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java

```diff
@@ -65,6 +69,22 @@ public class DBRecord implements Writable, DBWritable, Configurable {
    */
   protected List<ColumnType> columnTypes;
 
+  /**
+   * Need to cache column types to set fields of input record for where clause on {@link PreparedStatement}
+   * in the right order.
+   */
+  protected List<ColumnType> modifiableColumnTypes;
```

Comment:
what is the use of this field?

Reply:
In case of update, the no. of question marks which are to be written in the prepared statement throw Out of bounds exception as column types contains all the columns of the table. So in order to remove that exception, a new list is being created namely modifiableColumnTypes, which is getting updated to contain all the information about the column names on which the updation needs to take place.